internal/hook/wait_exec_hook_handler_test.go

```diff
@@ -0,0 +1,926 @@
+							Name: "container1",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						// container 2 is still waiting when the first hook executes in container1
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container2",
+							State: v1.ContainerState{
+								Waiting: &v1.ContainerStateWaiting{},
+							},
+						}).
+						Result(),
+				},
+				{
+					name: "my-hook-1",
+					hook: &velerov1api.ExecHook{
+						Container: "container2",
+						Command:   []string{"/usr/bin/bar"},
+					},
+					error: nil,
+					pod: builder.ForPod("default", "my-pod").
+						ObjectMeta(builder.WithResourceVersion("3")).
+						Containers(&v1.Container{
+							Name: "container1",
+						}).
+						Containers(&v1.Container{
+							Name: "container2",
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container1",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container2",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						Result(),
+				},
+			},
+			changes: []change{
+				// 1st modification: container1 starts running, resourceVersion 2, container2 still waiting
+				{
+					wait: 10 * time.Millisecond,
+					updated: builder.ForPod("default", "my-pod").
+						ObjectMeta(builder.WithResourceVersion("2")).
+						Containers(&v1.Container{
+							Name: "container1",
+						}).
+						Containers(&v1.Container{
+							Name: "container2",
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container1",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container2",
+							State: v1.ContainerState{
+								Waiting: &v1.ContainerStateWaiting{},
+							},
+						}).
+						Result(),
+				},
+				// 2nd modification: container2 starts running, resourceVersion 3
+				{
+					wait: 10 * time.Millisecond,
+					updated: builder.ForPod("default", "my-pod").
+						ObjectMeta(builder.WithResourceVersion("3")).
+						Containers(&v1.Container{
+							Name: "container1",
+						}).
+						Containers(&v1.Container{
+							Name: "container2",
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container1",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container2",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						Result(),
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			source := fcache.NewFakeControllerSource()
+			go func() {
+				source.Add(test.initialPod)
+				for _, change := range test.changes {
```

Comment:
Some comments around this loop, and maybe at the top of the test function, would be helpful to remember that the `changes` array is meant to be a set of versions of a given pod over time.